x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/components/fleet_server_on_prem_instructions.tsx

```diff
@@ -0,0 +1,761 @@
+        return false;
+      }
+    },
+    [setError]
+  );
+
+  const onSubmit = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      if (validate(fleetServerHost)) {
+        await addFleetServerHost(fleetServerHost);
+      }
+      setIsCalloutVisible(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [fleetServerHost, addFleetServerHost, validate]);
+
+  const onChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setFleetServerHost(e.target.value);
+      if (error) {
+        validate(e.target.value);
+      }
+    },
+    [error, validate, setFleetServerHost]
+  );
+
+  return {
+    title: i18n.translate('xpack.fleet.fleetServerSetup.addFleetServerHostStepTitle', {
+      defaultMessage: 'Add your Fleet Server host',
+    }),
+    status: undefined,
+    children: (
+      <EuiForm onSubmit={onSubmit}>
+        <EuiText>
+          <FormattedMessage
+            id="xpack.fleet.fleetServerSetup.addFleetServerHostStepDescription"
+            defaultMessage="Specify the URL your agents will use to connect to Fleet Server. This should match the public IP address or domain of the host where Fleet Server will run. By default, Fleet Server uses port `8220`."
+          />
+        </EuiText>
+        <EuiSpacer size="m" />
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFieldText
+              fullWidth
+              placeholder={'http://127.0.0.1:8220'}
+              value={fleetServerHost}
+              isInvalid={!!error}
+              onChange={onChange}
+              prepend={
+                <EuiText>
+                  <FormattedMessage
+                    id="xpack.fleet.fleetServerSetup.addFleetServerHostInputLabel"
+                    defaultMessage="Fleet Server host"
+                  />
+                </EuiText>
+              }
+            />
+            {error && <EuiFormErrorText>{error}</EuiFormErrorText>}
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton isLoading={isLoading} onClick={onSubmit}>
+              <FormattedMessage
+                id="xpack.fleet.fleetServerSetup.addFleetServerHostButton"
+                defaultMessage="Add host"
+              />
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        {isCalloutVisible && (
+          <>
+            <EuiSpacer size="m" />
+            <EuiCallOut
+              iconType="check"
+              size="s"
+              color="success"
+              title={
+                <FormattedMessage
+                  id="xpack.fleet.fleetServerSetup.addFleetServerHostSuccessTitle"
+                  defaultMessage="Added Fleet Server host"
+                />
+              }
+            >
+              <FormattedMessage
+                id="xpack.fleet.fleetServerSetup.addFleetServerHostSuccessText"
+                defaultMessage="Added {host}. You can edit your Fleet Server hosts in {fleetSettingsLink}."
```

Comment:
after clicking `Add host`, if you edit the input field the message updates with the edits (without saving) like this:
![image](https://user-images.githubusercontent.com/1965714/123712430-2621b700-d827-11eb-9702-97ed8e5c7d28.png)

we probably need to have a state that stores current user input vs saved user input